# Goal: triggers the controller.
int32 rod_arm
int32 surface_arm
float64 alpha
float64 goal_force
float64 vd_amplitude
float64 vd_frequency
float64 wd_amplitude
float64 wd_frequency
float64 nullspace_gain
float64 init_t_error
float64 init_k_error
bool use_nullspace
bool use_estimates

int32 LEFT_ARM = 0
int32 RIGHT_ARM = 1
---
# Result:

---
# Feedback: information that may be used to debug the controller
geometry_msgs/TwistStamped relative_twist
geometry_msgs/TwistStamped absolute_twist
geometry_msgs/TwistStamped controller_output
geometry_msgs/Vector3 r1
geometry_msgs/Vector3 r2
geometry_msgs/Vector3 p1
geometry_msgs/Vector3 p2
geometry_msgs/Vector3 pc
geometry_msgs/Vector3 pc_est
geometry_msgs/Vector3 t
geometry_msgs/Vector3 t_est
geometry_msgs/Vector3 k
geometry_msgs/Vector3 k_est
geometry_msgs/WrenchStamped surface_wrench
float64 task_compatibility
float64 alpha
float64 force_error
float64 force_d
float64 torque_error
float64 torque_d
float64 translational_angle_error
float64 rotational_angle_error
float64 pc_distance_error
float64 vs
float64 wr